migration/migration/src/lib.rs

```diff
@@ -108,8 +123,41 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 
+		/// Set Tokens (GAME & PLAY) balances with the same amount as native currency has
+		#[pallet::weight(5_000_000)]
+		pub fn tokens_airdrop(origin: OriginFor<T>) -> DispatchResult {
+			ensure_root(origin.clone()).is_ok();
+
+			if <TokensVersion<T>>::get() == StorageVersion::V2Imported {
+				return Ok(())
+			}
+			let mut accounts: u32 = 0;
+
+			for (acc_id, _acc_info) in <frame_system::Account::<T>>::iter() {				
+				// let bal: u128 = acc_info.data.free.saturated_into();	// --> no field `free` on type `AccountData`
+				let bal: u128 = <T as pallet_balances::Config>::AccountStore::get(&acc_id).free.saturated_into();
+				let balance: <T as orml_tokens::Config>::Balance = bal.saturated_into();
+				
+				let lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(acc_id.clone());
+				orml_tokens::Pallet::<T>::set_balance(
```

Comment:
`orml_currency`provides helper methods and adapters to work with token balances. Take a look at `update_balance` instead of calling nested extrinsics here?

Reply:
What about handling `TotalIssuance`, `existential deposit`, etc. I don't see `update_balance` providing this.